src/adapters.ts

```diff
@@ -128,16 +128,23 @@ export class Adapters implements LibAdapters {
                 }
             }
 
-            private adaptResults(query: PgQuery, rows: QueryResult) {
+            private adaptResults(query: PgQuery, res: QueryResult) {
                 if (query.rowMode) {
-                    throw new NotSupported('pg rowMode')
+                    throw new NotSupported('pg rowMode');
                 }
                 return {
-                    ...rows,
+                    ...res,
+                    // clone rows to avoid leaking symbols
+                    rows: res.rows.map(row => {
+                        return Object.entries(row).reduce((obj, [key, val]) => {
+                            obj[key] = val;
+                            return obj;
+                        }, {});
+                    }),
```

Comment:
Any reason not to use `res.rows.map(row => Object.fromEntries(Object.entries(row))` ?

Reply:
Nevermind, you answered that in your PR comment !